Automattic/wp-calypso
# GS on Personal A/B: Pattern Assembler to show correct copy when font and/or color are used

Related to https://github.com/Automattic/dotcom-forge/issues/2861

## Proposed Changes

* Propagate the `globalStylesInPersonalPlan` property to the necessary component in the `design-preview` package and made the necessary components GS A/B aware, they'll now return the corresponding text based on the required plan.
* Updated Calypso's `screen-color-palettes` and `screen-font-pairings` to be GS A/B aware and display the upsell text accordingly.

| Before | After |
-----|-----
![Captura de Pantalla 2023-07-05 a las 16 47 46](https://github.com/Automattic/wp-calypso/assets/1989914/be9dcdf6-8aff-418f-9a93-f8473719d97f)|![Captura de Pantalla 2023-07-05 a las 16 46 42](https://github.com/Automattic/wp-calypso/assets/1989914/195edc76-47d1-4105-8e2d-37a0795834a9)
![Captura de Pantalla 2023-07-05 a las 16 47 53](https://github.com/Automattic/wp-calypso/assets/1989914/3a02e0eb-0a9b-4ee2-b2cc-57d569802802)|![Captura de Pantalla 2023-07-05 a las 16 46 55](https://github.com/Automattic/wp-calypso/assets/1989914/64093f92-c075-4d5e-bdac-4223e1174da0)




## Testing Instructions



* Use Calypso live link
* Make sure you belong to the treatment group.
* Go through the /start process on a free plan until you reach the Pattern Assembler.
* Go to the bottom and use the assembler CTA.
* It should have a nav item that says Colors, click that and select a premium color.
* Just under the `Colors` title you should see a text that reads `Choose from our curated color palettes when you upgrade to the Personal plan or above.`
* Repeat for fonts.

You can change your group here (21268-explat-experiment), you also need to clear the cache when switching between groups by executing this command in your sandbox:
`wp_cache_delete( 'global-styles-on-personal-{siteId}', 'a8c_experiments' );` 
## Pre-merge Checklist

## Pre-merge Checklist
